src/support_enumeration.jl

```diff
@@ -0,0 +1,252 @@
+#=
+Compute all mixed Nash equilibria of a 2-player (non-degenerate) normal
+form game by support enumeration.
+
+References
+----------
+B. von Stengel, "Equilibrium Computation for Two-Player Games in
+Strategic and Extensive Form," Chapter 3, N. Nisan, T. Roughgarden, E.
+Tardos, and V. Vazirani eds., Algorithmic Game Theory, 2007.
+=#
+
+"""
+    support_enumeration(g::NormalFormGame)
+
+Compute mixed-action Nash equilibria with equal support size for a
+2-player normal form game by support enumeration. For a
+non-degenerate game input, these are all Nash equilibria.
+
+The algorithm checks all the equal-size support pairs; if the
+players have the same number n of actions, there are 2n choose n
+minus 1 such pairs. This should thus be used only for small games.
+
+# Arguments
+* `g::NormalFormGame`: NormalFormGame instance.
+
+# Returns
+* `::Vector{Tuple{Vector{Float64},Vector{Float64}}}`: Mixed-action
+    Nash equilibria that are found.
+"""
+function support_enumeration(g::NormalFormGame)
+
+    task = support_enumeration_task(g)
+
+    NEs = Tuple{Vector{Float64},Vector{Float64}}[NE for NE in task]
+
+    return NEs
+
+end
+
+"""
+    support_enumeration_task(g::NormalFormGame)
+
+Task version of `support_enumeration`.
+
+# Arguments
+* `g::NormalFormGame`: NormalFormGame instance.
+
+# Returns
+* `::Task`: runnable task for generating Nash equilibria.
+"""
+function support_enumeration_task(g::NormalFormGame)
+
+    N = length(g.nums_actions)
+    if N != 2
+        throw(ArgumentError("Implemented only for 2-player games"))
+    end
```

Comment:
I think @oyamad's suggestion is more Julian because it leave the door open to implement this routine for games with a different number of players.